src/components/molecules/SectionRenderer/SectionHeader.tsx

```diff
@@ -35,7 +36,14 @@ function SectionHeader(props: SectionHeaderProps) {
   const dispatch = useAppDispatch();
   const [isHovered, setIsHovered] = useState<boolean>(false);
 
-  const {NameDisplay, NamePrefix, NameSuffix, NameContext} = useSectionCustomization(sectionBlueprint.customization);
+  const {NameDisplay, NamePrefix, NameSuffix, NameContext, NameCounter} = useSectionCustomization(
+    sectionBlueprint.customization
+  );
+
+  const Counter: SectionCustomComponent = useMemo(
+    () => NameCounter.Component ?? SectionHeaderDefaultNameCounter,
```

Comment:
Curious to what you think about this @devcatalin. I can always revert this and do it conditionally as you can see in https://github.com/kubeshop/monokle/pull/1569/commits/d467cc016af1b5d1e283a7fe185a1b3af4b12534.

Reply:
This looks good!